Sources/WhisperKit/Core/WhisperKit.swift

```diff
@@ -192,45 +190,34 @@ public class WhisperKit: Transcriber {
                     callback(progress)
                 }
             }
-
-            let modelFolderName = modelFolder.appending(path: "openai_whisper-\(variant)")
-            return modelFolderName
+            return modelFolder.appending(path: "openai_whisper-\(variant)")
         } catch {
             Logging.debug(error)
             throw error
         }
     }
 
     /// Sets up the model folder either from a local path or by downloading from a repository.
-    public func setupModels(
+    public static func setupModelFolder(
         model: String?,
         downloadBase: URL? = nil,
         modelRepo: String?,
         modelFolder: String?,
-        download: Bool
-    ) async throws {
+        useBackgroundDownloadSession: Bool
+    ) async throws -> URL {
         // Determine the model variant to use
         let modelVariant = model ?? WhisperKit.recommendedModels().default
-
         // If a local model folder is provided, use it; otherwise, download the model
-        if let folder = modelFolder {
-            self.modelFolder = URL(fileURLWithPath: folder)
-        } else if download {
```

Comment:
This is needed for cased where someone wants to create the object but handle the downloading themselves. In fact I've been meaning to move this to the transcribe call so that the WhisperKit init is completely independent of any internet requests and purely sets up the object in all scenarios. This is important for folks that want to use the auxiliary functions of whisperkit by itself, without initializing models. Only when the developer shows intent to transcribe without any available models, it downloads as a fallback.

Reply:
I do like the idea of modelFolder being non-optional, but we can't rely on the download function directly, we should build "future" model folder that download would generate from other swift-transformer functions

Reply:
sounds good, seems more complex then I thought. Reverted back some changes, please take a 2nd look

Reply:
"Can the model now be used completely offline? It seems that loadTokenizer still needs to connect to hf (huggingface) to download.